main.go

```diff
@@ -91,17 +104,33 @@ func getModules() ([]*modEntry, error) {
 
 	for _, mod := range mods {
 		rev := mod.Version
+		url, err := vcs.RepoRootForImportPath(mod.Path, false)
+		if err != nil {
+			return nil, err
+		}
+
 		if commitShaRev.MatchString(rev) {
 			rev = commitShaRev.FindAllStringSubmatch(rev, -1)[0][1]
 		} else if commitRevV2.MatchString(rev) {
 			rev = commitRevV2.FindAllStringSubmatch(rev, -1)[0][1]
 		} else if commitRevV3.MatchString(rev) {
 			rev = commitRevV3.FindAllStringSubmatch(rev, -1)[0][1]
+		} else if mod.Path != url.Root {
+			subPath := strings.Split(mod.Path, url.Root+"/")[1]
+			if !versionNumber.MatchString(subPath) {
+				rev = subPath + "/" + rev
```

Comment:
Why module path prefix is in `else if` branch? I think we should handle it as long as the version string is not git sha style?

Reply:
This handles dependencies like ``github.com/Azure/go-autorest/autorest/azure/auth v0.5.1``. If you have a nicer way please suggest it.

Reply:
> This handles dependencies like `github.com/Azure/go-autorest/autorest/azure/auth v0.5.1`. If you have a nicer way please suggest it.

Does the `rev` in case `commitRevV3` need to prepend module path like this?

Reply:
I am not sure. Do you have an example?